lib/migration-chunks/validation/duplicate-props.js

```diff
@@ -0,0 +1,52 @@
+const errorMessages = require('./errors');
+
+const isSameEntity = (step, otherStep) => {
+  return (
+    step.payload.contentTypeId === otherStep.payload.contentTypeId &&
+    step.payload.fieldId === otherStep.payload.fieldId
+  );
+};
+
+const hasProps = (step) => step.payload && step.payload.props;
+
+const getErrorMsgParams = (step) => {
+  const type = step.type.split('/')[0];
+  const params = {
+    field: {
+      type: 'field',
+      id: step.payload.fieldId
+    },
+    contentType: {
+      type: 'content type',
+      id: step.payload.contentTypeId
+    }
+  };
+  return params[type];
+};
+
+module.exports = (chunks) => {
+  return chunks.reduce((errors, chunk) => {
+    const chunkErrors = chunk.reduce((chunkErrors, step, i) => {
```

Comment:
Not too sure about this, but it took me a minute to grasp what the double-nested `reduce` does.
Wondering whether it makes it more readable, if we pull the reducer into its own fn, but maybe it just makes you jump between the fns more.